infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -710,15 +707,19 @@ private void handleAllocation(RequestMsg req, ChannelHandlerContext ctx, IServer
 
         // update the cache of conflict parameters
         if (tokenRequest.hasTxnResolution()) {
-            tokenRequest.getTxnResolution().getWriteConflictParamsSetList()
-                    .forEach((item) -> {
-                        // insert an entry with the new timestamp using the
-                        // hash code based on the param and the stream id.
-                        item.getValueList().forEach(conflictParam ->
-                                cache.put(new ConflictTxStream(getUUID(item.getKey()),
-                                        conflictParam.toByteArray(), newTail - 1)));
-                    });
+            Set<ConflictTxStream> conflictTxStreamList = tokenRequest.getTxnResolution()
+                    .getWriteConflictParamsSetList()
+                    .stream()
+                    .flatMap(item -> {
+                        final UUID streamId = getUUID(item.getKey());
+                        return item.getValueList()
+                                .stream()
+                                .map(conflictParam -> new ConflictTxStream(streamId, conflictParam.toByteArray()));
+                    }).collect(Collectors.toCollection(LinkedHashSet::new));
+
+           cache.put(conflictTxStreamList, newTail - 1);
```

Comment:
This seems incorrect, the new tail is computed as following `long newTail = globalLogTail + tokenRequest.getNumTokens();` , but the subsequent logic assumes that `tokenRequest.getNumTokens` is strictly 1 and therefore uses `newTail - 1` as the token.

Essentially, if the client requests multiple tokens, the conflict parameters will only be placed in the last token, it seems like this is more of a coincidence than design. I would suggest making these calculations more strict.

Reply:
Since a client only ever requests one token when they go to commit, I can remove the deprecated APIs, update this part of the code and relevant tests in a subsequent PR.

Reply:
Sounds good.